--- ./build/config/rust.gni.orig
+++ ./build/config/rust.gni
@@ -198,13 +198,13 @@
 rust_abi_target = ""
 if (is_linux || is_chromeos) {
   if (current_cpu == "arm64") {
-    rust_abi_target = "aarch64-unknown-linux-gnu"
+    rust_abi_target = "aarch64-unknown-linux-musl"
     cargo_target_abi = ""
   } else if (current_cpu == "x86") {
-    rust_abi_target = "i686-unknown-linux-gnu"
+    rust_abi_target = "i586-unknown-linux-musl"
     cargo_target_abi = ""
   } else if (current_cpu == "x64") {
-    rust_abi_target = "x86_64-unknown-linux-gnu"
+    rust_abi_target = "x86_64-unknown-linux-musl"
     cargo_target_abi = ""
   } else if (current_cpu == "arm") {
     if (arm_float_abi == "hard") {
@@ -214,18 +214,18 @@
     }
     if (arm_arch == "armv7-a" || arm_arch == "armv7") {
       # No way to inform Rust about the -a suffix.
-      rust_abi_target = "armv7-unknown-linux-gnueabi" + float_suffix
+      rust_abi_target = "armv7-unknown-linux-musleabi" + float_suffix
       cargo_target_abi = "eabi" + float_suffix
     } else {
-      rust_abi_target = "arm-unknown-linux-gnueabi" + float_suffix
+      rust_abi_target = "armv6-unknown-linux-musleabi" + float_suffix
       cargo_target_abi = "eabi" + float_suffix
     }
   } else if (current_cpu == "riscv64") {
-    rust_abi_target = "riscv64gc-unknown-linux-gnu"
+    rust_abi_target = "riscv64-unknown-linux-musl"
     cargo_target_abi = ""
   } else {
     # Best guess for other future platforms.
-    rust_abi_target = current_cpu + "-unknown-linux-gnu"
+    rust_abi_target = current_cpu + "-unknown-linux-musl"
     cargo_target_abi = ""
   }
 } else if (is_android) {

